### DISCUSSION TITLE
Can i model a cable driven manipulator with SOFA?

### DISCUSSION THREAD

**Discussion by shimaamiri** [sofa-framework/sofa#4653] [Ideas / Suggestions]
Hello,
I'm going to model a cable-driven serial manipulator in an interface which can be integrated with ROS like Gazebo. Modeling cable in gazebo is challenging but i saw cables can be modeled properly with SOFA. 
So I wanted to know is it possible to model my whole manipulator with SOFA? i would appreciate if you have any other suggestions.

**Comment by damienmarchal** [comment]
@shimaamiri 

At DEFROST we modeled several cable driven robots including manipulator as well as having tried different way to integrate it with ROS or GUI for interactive control. So I'm quite confident it is possible. 

I don't know if this will help, it will depend on your design, but here are examples of coupling deformable/rigid parts. 
- https://github.com/SofaDefrost/TIRREX.Endoscopy
- https://github.com/SofaDefrost/TIRREX.XXL
- https://github.com/SofaDefrost/SoftRigidHybridArm

There is also example of cable actuation in the SoftRobots plugin.
